### PR TITLE
Fix for sonarcloud bugs

### PR DESCRIPTION
**Pull request by stefan-andritoiu**

### CODE REVIEW

File: src/iio/iio.c [hunk 1]
```diff
@@ -129,16 +129,20 @@ mraa_iio_get_channel_data(mraa_iio_context dev)
                 fd = open(buf, O_RDONLY);
                 if (fd != -1) {
                     read(fd, readbuf, 31 * sizeof(char));
+
```
Comment (alext-mkrs) [inline src/iio/iio.c]: Please don't add unnecessary formatting changes, those just pollute the git history.

File: src/iio/iio.c [hunk 2]
```diff
@@ -172,10 +176,17 @@ mraa_iio_get_channel_data(mraa_iio_context dev)
     }
     closedir(dir);
 
-    // channel location has to be done in channel index order so do it afetr we
+    // channel location has to be done in channel index order so do it after we
```
Comment (alext-mkrs) [inline src/iio/iio.c]: Please don't add unnecessary formatting changes, those just pollute the git history. This one is legitimate, but please split it out to a separate commit.

File: src/iio/iio.c [hunk 3]
```diff
@@ -172,10 +176,17 @@ mraa_iio_get_channel_data(mraa_iio_context dev)
...
     // have grabbed all the correct info
     for (i = 0; i < dev->chan_num; i++) {
-	chan = &dev->channels[i];
+	    chan = &dev->channels[i];
```
Comment (alext-mkrs) [inline src/iio/iio.c]: Incorrect text alignment - looks like tabs and spaces got mixed on one line, please double-check.

File: src/iio/iio.c [hunk 4]
```diff
@@ -172,10 +176,17 @@ mraa_iio_get_channel_data(mraa_iio_context dev)
...
+
+        if(chan->bytes <= 0)
```
Comment (alext-mkrs) [inline src/iio/iio.c]: This one looks good to me, but please split it out to a separate commit, so that each commit is a logically contained change.

File: src/gpio/gpio.c [hunk 5]
```diff
@@ -68,7 +68,10 @@ mraa_gpio_close_event_handles_sysfs(int fds[], int num_fds)
     }
 
     for (int i = 0; i < num_fds; ++i) {
-        close(fds[i]);
+        // Check required to avoid closing stdin and of an uninitialized fd
+        if(fds[i] != 0) {
```
Comment (alext-mkrs) [inline src/gpio/gpio.c]: Haven't noticed this one before, sorry - please add a space between `if` and `(` to align with mraa code style.

**Comment by Propanu** [comment]
Merged and ran clang-format address issues raised by @alext-mkrs. Thank you!
